lib/file.coffee

```diff
@@ -67,3 +67,6 @@ class File
 
   isPathEqual: (pathToCompare) ->
     @path is pathToCompare or @realPath is pathToCompare
+
+  # Return a snapshot of the instance's filesystem properties for API consumption
+  getMetadata: -> {@symlink, @realPath, @status, @stats}
```

Comment:
Actually never mind. I've lost patience during all this bikeshedding. I'll kill the second argument and write whatever craploads more code I need to write in order to get things off the ground. Seriously.